server/src/com/thoughtworks/go/server/controller/PluginController.java

```diff
@@ -0,0 +1,89 @@
+/*************************GO-LICENSE-START*********************************
+ * Copyright 2014 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *************************GO-LICENSE-END***********************************/
+
+package com.thoughtworks.go.server.controller;
+
+import com.thoughtworks.go.plugin.api.request.DefaultGoPluginApiRequest;
+import com.thoughtworks.go.plugin.api.response.DefaultGoApiResponse;
+import com.thoughtworks.go.plugin.api.response.GoPluginApiResponse;
+import com.thoughtworks.go.plugin.infra.PluginManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.View;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.HashMap;
+import java.util.Map;
+
+@Controller
+public class PluginController {
+    private PluginManager pluginManager;
+
+    @Autowired
+    public PluginController(PluginManager defaultPluginManager) {
+        this.pluginManager = defaultPluginManager;
+    }
+
+    @RequestMapping(value = "/{pluginId}/{requestName}", method = RequestMethod.GET)
+    public ModelAndView handleAboutRequest(
+            @PathVariable String pluginId,
+            @PathVariable String requestName,
+            HttpServletRequest request) {
+        DefaultGoPluginApiRequest apiRequest = new DefaultGoPluginApiRequest(null, null, requestName);
+        apiRequest.setRequestParams(getParameterMap(request));
```

Comment:
I don't think we should send the whole map. It should be a structured (known) subset (or a subset which is prefixed by the plugin ID or something). Sometimes there's a lot of information that is there in it.

If the information there is only coming from the login page which the plugin controls, then it's fine. If there is other info that jetty puts there, we might want to look at getting rid of it before sending to the plugin.